catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityCheckerPlugin.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package ddf.catalog.metacard.validation;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.filter.FilterAdapter;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PreQueryPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.source.UnsupportedQueryException;
+
+public class MetacardValidityCheckerPlugin implements PreQueryPlugin {
+
+    protected static FilterBuilder filterBuilder;
+    protected static FilterAdapter filterAdapter;
+
+    public MetacardValidityCheckerPlugin(FilterBuilder filterBuilder, FilterAdapter filterAdapter) {
+        MetacardValidityCheckerPlugin.filterBuilder = filterBuilder;
+        MetacardValidityCheckerPlugin.filterAdapter = filterAdapter;
+    }
+
+    @Override
+    public QueryRequest process(QueryRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        QueryImpl query;
+        try {
+            if (!filterAdapter.adapt(input.getQuery(), new ValidityParamDelegate())) {
+                query = new QueryImpl(filterBuilder.allOf(input.getQuery(),
+                        filterBuilder.attribute(Metacard.VALID_STATE).is().equalTo().bool(true)));
+            } else {
+                query = new QueryImpl(input.getQuery());
+            }
+        } catch (UnsupportedQueryException e) {
+            throw new StopProcessingException(e.getMessage());
```

Comment:
I assume you're rethrowing `e.getMessage()` to avoid leaking implementation details. If that's the case, then we usually also log the full exception. Makes it a lot easier to debug down the line.